Maryia/91511/fix: verify_email counter in withdrawal

maryia-deriv opened the pull request:
## Changes:

Please include a summary of the change and which issue is fixed below:
-   to ensure Didn't Receive an email is shown before counter starts
-   to start verify_email counter in withdrawal only after second request
-   to store verify_email data in sessionStorage

## When you need to add unit test

-   [ ] If this change disrupt current flow
-   [ ] If this change is adding new flow

## When you need to add integration test

-   [ ] If components from external libraries are being used to define the flow, e.g. @deriv/components
-   [ ] If it relies on a very specific set of props with no default behavior for the current component.

## Test coverage checklist (for reviewer)

-   [ ] Ensure utility / function has a test case 
-   [ ] Ensure all the tests are passing

## Type of change

-   [x] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

farzin-deriv reviewed (commented):
1. Do we really need to add so much logic on the client side just for sending a request? :(

2. We are moving from huge store files to small reusable hooks for better maintenance and readability so we can scale (#6339), After changes in (#7218) the `useVerifyEmail` hook is not reusable and it contains lots of logic which make it hard to understand for a new joiner, Could you please break it into smaller reusable hooks instead 🙏🏻 

3. We are making more dependency on `ClientStore` just to get and set data from `sessionStorage`, Since we are trying to remove these dependencies between stores this will create more work for us, Can we instead just create a new reusable hook for writing and reading from `sessionStorage` instead and remove the dependency to `ClientStore` 🙏🏻 

4. Since the `useVerifyEmail` hook is doing too many things now and it's not just a wrapper around the `useWS`, The test we have for the `useWS` doesn't guarantee this hook outcome, Would you please help to write tests for the `useVerifyEmail` hook 🙏🏻

maryia-deriv commented:
@farzin-deriv 
1. useVerifyEmail already had a lot of logics inside it when I started to work on it. It was returning an object which contains the data about when verify_email was sent last time and how many times, etc. and it continues to return this object but now it's improved, this data are not getting lost anytime Withdrawal tab is closed anymore or when the page is reloaded.
2. The small reusable useVerifyEmail hook that you created was only working locally before I changed it, that's why counter was getting reset every time a user switches to another tab. With my change it's still reusable for any other verify_email requests. I added caching of data about when verify_email was sent last time and how many times in the global state (mobx client-store) in order to save the data about this counter outside of Withdrawal tab. The data are stored under the key for each type of verify_email request, so it's still reusable.
3. No dependencies between stores are created by my change, I simply store the data about when verify_email request that I mentioned above in sessionStorage instead of localStorage to make sure that the counter data are cleared and the counter is not displayed when a new tab with Withdrawal page is opened after a user is redirected to it after clicking verification link. We are already setting a lot of values to localStorage that we need to store in localStorage from client-store for better user experience so that data are not lost when a user reloads the page. That's why I disagree with you that we cannot use sessionStorage for the purpose I explained.
4. I will surely update the test for this hook. Thank you for pointing this out👍🏻
